src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -816,6 +817,19 @@ def sql_query(self) -> str:
                 else snowflake.snowpark._internal.utils.EMPTY_STRING
             )
             self._sql_query = f"{analyzer_utils.SELECT}{self.projection_in_str}{analyzer_utils.FROM}{from_clause}{where_clause}{order_by_clause}{limit_clause}{offset_clause}"
+
+            for plan in self.analyzer.subquery_plans:
+                for query in plan.queries[:-1]:
```

Comment:
also, if we look at the with_subqueries implementation in the analyzer.py it also handles the schema query, we will need to do the same thing, i think we might still need to do that at the analyzer level and just call the same function to propogate the field properly

Reply:
The original reason why we handle `with_subqueries` is handle the condition when
- current plan's schema query = a subquery in subquery_plans

This does not apply to select statement since subquery plans for us can only come from filter expressions.

Reply:
so here is what i see how plan node with subquery works today, snowpark does resolve for every intermediate step, however, for node with subquery, it is not correct after resolve, https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/analyzer/analyzer.py#L780

this corrects the post-actions, queries as well as the schema queries. For select statement, it seems the same to me, where we also need to correct the schema query, otherwise the schema query will not be valid because it refers to a table that is not created. for example, following is the schema query for the select Statement 
```
 SELECT NOT "A" IN (( SELECT "A" FROM ( SELECT  *  FROM (SNOWPARK_TEMP_TABLE_3QWYPIQ4JV)) WHERE ("A" < 2))) AS "IN_RESULT" FROM ( SELECT "A", "B", "C", "D" FROM ( SELECT $1 AS "A", $2 AS "B", $3 AS "C", $4 AS "D" FROM  VALUES (0 :: BIGINT, 'a' :: STRING, 0 :: BIGINT, 0 :: BIGINT)))
```

as you can see it is referring to a SNOWPARK_TEMP_TABLE_3QWYPIQ4JV that is not defined anywhere.

So the selectstatement should be handled in the same way as how we handle the resolved snowflakeplane  (because selectstatement its-self treated as a resolved node), the problem today seems that we only handled the resolved SnowflakePlan, but not the selectStatement when the logical plan passed is a selectStatement here. Therefore, other than handling resolved SnowflakePlan here https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/analyzer/analyzer.py#L781, we should also handle the logical plan if it is a SelectStatement. The resolve step will record all subquery resolved